### ISSUE TITLE
Task IM-048: Implement CategoricalNB baseline model

### ISSUE THREAD

**Issue by cliu238** [cliu238/comsa_claude#22]
## Task Overview
- **Task ID**: IM-048
- **Priority**: Medium
- **Dependencies**: VADataProcessor, numeric encoding
- **Description**: Implement CategoricalNB baseline model with appropriate missing data handling

## Current Context
We have successfully implemented:
1. XGBoost baseline model (IM-045) with sklearn interface
2. Random Forest baseline model (IM-046) with feature importance
3. Logistic Regression baseline model (IM-047) with regularization
4. InSilicoVA model (IM-013) with Docker integration
5. Model comparison framework (IM-035) for systematic evaluation

### 1. Model Architecture
- Create `baseline/models/categorical_nb_model.py` with sklearn-compatible interface
- Create `baseline/models/categorical_nb_config.py` for configuration management
- Use sklearn's CategoricalNB for categorical feature handling
- Implement proper missing data handling strategies

### 2. Key Features
- **Categorical Data Handling**:
  - Native support for categorical features
  - No need for one-hot encoding
  - Handles "Y"/"." format from OpenVA
- **Missing Data Strategies**:
  - Option to skip missing values
  - Option to treat missing as separate category
  - Option to impute with mode
- **Smoothing Parameters**:
  - Additive (Laplace) smoothing
  - Min category frequency threshold
- **Class Balancing**:
  - Support class_weight for imbalanced data
  - Prior probability adjustment

### 3. Model Interface (following existing patterns)
```python
class CategoricalNBModel(BaseEstimator, ClassifierMixin):
    def __init__(self, config: Optional[CategoricalNBConfig] = None)
    def fit(self, X: pd.DataFrame, y: pd.Series) -> "CategoricalNBModel"
    def predict(self, X: pd.DataFrame) -> np.ndarray
    def predict_proba(self, X: pd.DataFrame) -> np.ndarray
    def get_feature_importance(self) -> pd.DataFrame  # Based on feature probabilities
    def calculate_csmf_accuracy(self, y_true: pd.Series, y_pred: np.ndarray) -> float
    def cross_validate(self, X: pd.DataFrame, y: pd.Series, cv: int = 5) -> Dict[str, float]
```

### 4. Configuration Schema
```python
class CategoricalNBConfig(BaseModel):
    alpha: float = 1.0  # Additive smoothing parameter
    force_alpha: bool = True  # If False, alpha is per-feature
    fit_prior: bool = True  # Learn class prior probabilities
    class_prior: Optional[List[float]] = None  # Custom priors
    min_categories: int = 1  # Min samples per category
    handle_missing: Literal["skip", "category", "mode"] = "category"
    categorical_features: Optional[List[str]] = None  # Auto-detect if None
    random_state: int = 42
    n_jobs: Optional[int] = None
```

### 5. Missing Data Handling
- **"skip"**: Ignore samples with missing values during training
- **"category"**: Treat missing as separate category (recommended for VA)
- **"mode"**: Impute with most frequent value per feature

### 6. Feature Importance
- Based on conditional probabilities P(feature|class)
- Information gain per feature
- Mutual information scores
- Compatible with model comparison framework

### 7. Testing Requirements
- Unit tests with >95% coverage
- Test all missing data handling strategies
- Test with pure categorical data
- Test with mixed numeric/categorical (should warn)
- Test feature importance extraction
- Test CSMF accuracy calculation
- Integration with model comparison framework

### 8. Integration Points
- Add to `baseline/models/__init__.py`
- Update model comparison scripts
- Ensure compatibility with OpenVA encoding
- Add to parallel execution framework

### 9. Documentation
- Comprehensive docstrings
- Example usage with VA data
- Performance characteristics
- Comparison with other baseline models

## Success Criteria
1. ✅ Sklearn-compatible interface matching other models
2. ✅ Native categorical feature handling
3. ✅ Robust missing data strategies
4. ✅ Feature importance based on probabilities
5. ✅ CSMF accuracy metric implementation
6. ✅ >95% test coverage
7. ✅ Integration with model comparison framework
8. ✅ Handles OpenVA "Y"/"." format natively
9. ✅ Clear documentation and examples

## Expected Outcomes
- Fast training on categorical VA data
- No preprocessing needed for categorical features
- Good performance when independence assumption holds
- Interpretable probability-based predictions
- Robust to missing data common in VA surveys

## VA-Specific Advantages
- Handles symptom presence/absence naturally
- Missing symptoms treated appropriately
- Fast inference for real-time applications
- Probabilistic interpretation aligns with VA uncertainty

## Next Task After Completion
After IM-048, the next logical task would be:
- IM-049: Implement InterVA model integration (high priority)
- IM-036: Create unified model comparison pipeline

**Comment by cliu238** [comment]
Task completed successfully with PR #31. CategoricalNB baseline model has been implemented and merged into main branch.
